python/setup.py

```diff
@@ -174,16 +193,22 @@ def run(self):
       sys.argv.remove(warnings_as_errors)
 
     # C++ implementation extension
-    ext_module_list.append(
+    ext_module_list.extend([
         Extension(
             "google.protobuf.pyext._message",
             glob.glob('google/protobuf/pyext/*.cc'),
             include_dirs=[".", "../src"],
-            libraries=['protobuf'],
+            libraries=libraries,
+            extra_objects=extra_objects,
             library_dirs=['../src/.libs'],
             extra_compile_args=extra_compile_args,
-        )
-    )
+        ),
+        Extension(
+            "google.protobuf.internal._api_implementation",
```

Comment:
I think this is only needed because of -DPROTOBUF_PYTHON_ALLOW_OVERSIZE_PROTOS? But I thought we agreed it was better to have you specify this via an API rather than the compile-time option that would apply to all users?